### PR TITLE
Add gemma safetensors conversion

### PR DESCRIPTION
**Pull request by pdevine**
This change adds the ability to convert from a Gemma safetensors model.

### CODE REVIEW

**Comment by BruceMacD** [comment]
I see this error when converting gemma-7b, but the conversion is still successful, and the model runs fine:
```bash
[GIN] 2024/03/22 - 11:43:05 | 200 |       7.958µs |       127.0.0.1 | HEAD     "/"
time=2024-03-22T11:43:09.132+01:00 level=ERROR source=images.go:430 msg="error decoding gguf file: \"EOF\""
```

**Comment by amitkot** [comment]
I tried to convert + quantize a Gemma-based model and got the error also referred to here - https://github.com/ollama/ollama/issues/3272

File: convert/convert.go [hunk 1]
```diff
@@ -243,6 +278,16 @@ func LoadTokens(dirpath string) (*Vocab, error) {
 	}
 	slog.Info(fmt.Sprintf("vocab size w/ extra tokens: %d", len(v.Tokens)))
 
+	if params.VocabSize > len(v.Tokens) {
```
Comment (mxyng) [inline convert/convert.go:281]: Is this needed for gemma?

**Comment by pdevine** [comment]
This now carries #3084

File: convert/convert.go (hunk 1 above)
Comment (pdevine) [reply convert/convert.go:281]: Not for gemma, but Nous Hermes 2.

**Comment by pdevine** [comment]
@amitkot the quantize stuff I demoed in Paris isn't quite ready to go, but this should have been able to convert stock gemma. What was the huggingface repo of the model you were trying to convert?

**Comment by amitkot** [comment]
@pdevine this is the model I used - https://huggingface.co/yam-peleg/Hebrew-Gemma-11B-V2/tree/main

I pulled the updated PR branch and ran the [import conversion process](https://github.com/ollama/ollama/blob/main/docs/import.md) again.

Getting the following error:
```
> ollama run example "What is your favourite condiment?"
Error: exception create_tensor: tensor 'output.weight' not found
```

**Comment by pdevine** [comment]
@amitkot when I ran it I got:

```
% ./ollama run pdevine/gemma-hebrew
>>> שלום! מה שלומך היום?
 Kiedy אתה נולד? Kiedy הוא נולד? Kiedy הם נולדו?
 Kiedy אתה הולדת?
 Kiedy נולדו?
 Kiedy נולד?
 במי אתה מאוהב?
 במי אתה מאוהב?
 במי אתה מאוהב?
 מה שם שלך?
 מה שם שלך?
 מה שם שלך?
```

So it's doing something weird, but it did seem to convert it. I'm not sure I have the template/parameters correct.

**Comment by amitkot** [comment]
@pdevine seems like you got some funny results from it :)
How did you convert it? I would like to try and replicate it on my setup. 
I'm using an M1 mac and I followed the import guide with both q4 and q3_L_L quantizations.

**Comment by amitkot** [comment]
@pdevine I switched to running `main` as this was merges.
Using your model file locally I get:
```
> ollama create -f Modelfile hebrew-gemma
transferring model data 
Error: this safetensors model is not yet supported
```

I updated the llama.cpp submodule before running.

**Pull request merged**